### PR TITLE
Add database access getting started guide

### PR DESCRIPTION
**Pull request by r0mant**
In preparation for the first alpha release with database access, I've added a short-ish getting started guide. Existing database access documentation is more of a reference as it's pretty extensive with a lot of details that makes it hard to follow for new users.

I also noticed that Sasha submitted https://github.com/gravitational/teleport/pull/5305 so tried to follow suggestions from there as well, I'm open to tweaking it further if needed.

### CODE REVIEW

File: docs/5.0/preview/database-access.md [hunk 1]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+
+### Step 1/3. Setup Aurora
+
+The first step is to configure an Aurora database to support IAM authentication
+which Teleport will use when connecting to it.
+
+If you don't have a database provisioned yet, create an instance of a PostgreSQL
+compatible Aurora instance in the [RDS control panel](https://us-west-1.console.aws.amazon.com/rds/home).
+Make sure to check "Password and IAM database authentication" in the Database
+Authentication dialog to enable IAM auth.
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:54]:
```suggestion
Authentication dialog.
```

File: docs/5.0/preview/database-access.md [hunk 2]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+
+For existing Aurora databases, the status of IAM authentication is displayed on
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:56]:
```suggestion
For existing Aurora instances, the status of IAM authentication is displayed on
```
?

File: docs/5.0/preview/database-access.md [hunk 3]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+```sql
+CREATE USER alice;
+GRANT rds_iam TO alice;
+```
+
+See a more detailed description of these steps in the [reference](#aws-rdsaurora-postgresql) below.
+
+### Step 2/3. Setup Teleport
+
+Head over to Teleport [downloads page](https://goteleport.com/teleport/download)
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:91]:
```suggestion
Download the latest version of Teleport for your platform from [downloads page](https://goteleport.com/teleport/download)
```

File: docs/5.0/preview/database-access.md [hunk 4]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+and download the latest version of Teleport for your platform. Database Access
+is available starting from `6.0.0-alpha.1` release.
+
+Start Teleport using the following command and point it to your Aurora database.
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:95]:
```suggestion
Start Teleport using the following command and point it to your Aurora database instance.
```

File: docs/5.0/preview/database-access.md [hunk 5]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+  --roles=proxy,auth,database \
+  --db-name=aurora \
+  --db-protocol=postgres \
+  --db-uri=postgres-aurora-instance-1.abcdefghijklm.us-west-1.rds.amazonaws.com:5432 \
+  --db-aws-region=us-west-1
+```
+
+!!! note "AWS credentials"
+    The node where the Teleport process is started should have AWS credentials
+    configured with the policy created above.
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:109]: nit: make the section referable from here and use a link?

File: docs/5.0/preview/database-access.md [hunk 6]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+
+Create a Teleport user that is allowed to connect to a particular database
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:111]:
```suggestion
Create a Teleport user that is allowed to connect to a particular database (e.g. `postgres`)
```

File: docs/5.0/preview/database-access.md [hunk 7]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+
+```shell
+sudo tctl users add alice root \
+  --db-names=postgres \
+  --db-users=alice
+```
+
+### Step 3/3. Connect to Database
+
+Now that Aurora is configured with IAM authentication, Teleport is started and
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:122]:
```suggestion
Now that Aurora is configured with IAM authentication, Teleport is running and
```

File: docs/5.0/preview/database-access.md [hunk 8]
```diff
@@ -38,6 +38,130 @@ With Database Access users can:
...
+the local user is created, we're ready to connect to the database.
+
+Log into Teleport with the user we've just created.
+
+!!! note "Teleport certificate"
+    For the ease of this guide, we're using an `--insecure` flag to accept
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:128]:
```suggestion
    For simplicity we're using an `--insecure` flag to accept
```

**Review by a-palchikov — APPROVED** [review]

**Comment by r0mant** [comment]
@klizhentas @benarent @webvictim Can you folks take a look?

File: docs/5.0/preview/database-access.md [hunk 9]
```diff
@@ -38,6 +38,134 @@ With Database Access users can:
...
+2. Download and install Teleport and connect it to the Aurora database.
+3. Connect to the Aurora database via Teleport.
+
+### Step 1/3. Setup Aurora
+
+In order to allow Teleport connections to an Aurora instance, it needs to support
+IAM authentication.
+
+If you don't have a database provisioned yet, create an instance of an Aurora
+PostgreSQL instance in the [RDS control panel](https://console.aws.amazon.com/rds/home).
```
Comment (benarent) [inline docs/5.0/preview/database-access.md:56]: For a fresh DB you need to select `Standard create` to enable Password and IAM

File: docs/5.0/preview/database-access.md [hunk 10]
```diff
@@ -38,6 +38,134 @@ With Database Access users can:
...
+Make sure to enable "Password and IAM database authentication" in the Database
+Authentication dialog.
+
+For existing Aurora instances, the status of IAM authentication is displayed on
+the Configuration tab and can be enabled by modifying the database instance.
+
+Next, create the following IAM policy attached to a user whose credentials a
```
Comment (benarent) [inline docs/5.0/preview/database-access.md:63]: I tried this policy and it's saying it doesn't have any permissions?

image

File: docs/5.0/preview/database-access.md [hunk 11]
```diff
@@ -38,6 +38,134 @@ With Database Access users can:
...
+{
+   "Version": "2012-10-17",
+   "Statement": [
+      {
+         "Effect": "Allow",
+         "Action": [
+             "rds-db:connect"
+         ],
+         "Resource": [
+             "arn:aws:rds-db:<region>:<account-id>:dbuser:<db-cluster-resource-id>/*"
```
Comment (benarent) [inline docs/5.0/preview/database-access.md:76]: In this resource example, is `dbuser` important? For default should this be changed to `postgres` ?

Comment (r0mant) [reply docs/5.0/preview/database-access.md:76]: @benarent No, this is a constant string so it should appear verbatim `dbuser`.

File: docs/5.0/preview/database-access.md [hunk 12]
```diff
@@ -38,6 +38,134 @@ With Database Access users can:
...
+         ]
+      }
+   ]
+}
+```
+
+Finally, connect to the database and create a database account with IAM auth
```
Comment (benarent) [inline docs/5.0/preview/database-access.md:83]: I would add a link to https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ConnectToPostgreSQLInstance.html for newbie Postgres users

File: docs/5.0/preview/database-access.md (hunk 11 above)
Comment (benarent) [reply docs/5.0/preview/database-access.md:76]: This was due to an error of trying to connect to `rds` vs `rds-db` marking as closed.

File: docs/5.0/preview/database-access.md (hunk 12 above)
Comment (benarent) [reply docs/5.0/preview/database-access.md:83]: I found https://aws.amazon.com/blogs/database/using-iam-authentication-to-connect-with-pgadmin-amazon-aurora-postgresql-or-amazon-rds-for-postgresql/ content also helpful as I learnt more about the specifics of IAM and Aurora Postgres

File: docs/5.0/preview/database-access.md (hunk 9 above)
Comment (r0mant) [reply docs/5.0/preview/database-access.md:56]: I've clarified this below.

File: docs/5.0/preview/database-access.md (hunk 12 above)
Comment (r0mant) [reply docs/5.0/preview/database-access.md:89]: Added a link below.

File: docs/5.0/preview/database-access.md [hunk 13]
```diff
@@ -38,6 +38,144 @@ With Database Access users can:
...
+Log into Teleport with the user we've just created.
+
+!!! note "Teleport certificate"
+    For simplicity, we're using an `--insecure` flag to accept Teleport's
+    self-signed certificate. For production usage make sure to configure proxy
+    with a proper certificate/key pair. See Teleport's general
+    [quickstart guide](../quickstart.md#step-1c-configure-domain-name-and-obtain-tls-certificates-using-lets-encrypt).
+
+```shell
+tsh login --insecure --proxy=localhost:3080 --user=alice
```
Comment (benarent) [inline docs/5.0/preview/database-access.md:148]: We should add a note that during beta customers will need to obtain `tsh` version 6.0

Comment (r0mant) [reply docs/5.0/preview/database-access.md:148]: Right, added above.

File: docs/5.0/preview/database-access.md [hunk 14]
```diff
@@ -38,6 +38,144 @@ With Database Access users can:
...
+2. Download and install Teleport and connect it to the Aurora database.
+3. Connect to the Aurora database via Teleport.
+
+### Step 1/3. Setup Aurora
+
+In order to allow Teleport connections to an Aurora instance, it needs to support
+IAM authentication.
+
+If you don't have a database provisioned yet, create an instance of an Aurora
+PostgreSQL instance in the [RDS control panel](https://console.aws.amazon.com/rds/home).
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:56]:
```suggestion
PostgreSQL in the [RDS control panel](https://console.aws.amazon.com/rds/home).
```
there's already an `instance` in the same sentence.

File: docs/5.0/preview/database-access.md [hunk 15]
```diff
@@ -38,6 +38,144 @@ With Database Access users can:
...
+         ]
+      }
+   ]
+}
+```
+
+!!! note "Resource ID"
+    Database resource ID is shown on the Configuration tab of a particular
+    database instance in RDS control panel, under "Resource id". For regular
+    RDS database it starts with `db-` prefix. For Aurora use database cluster
```
Comment (a-palchikov) [inline docs/5.0/preview/database-access.md:86]:
```suggestion
    RDS database it starts with `db-` prefix. For Aurora, use the database cluster
```

**Review by a-palchikov — APPROVED** [review]

**Comment by r0mant** [comment]
@benarent @webvictim Addressed Ben's last comments, anything else folks?

**Review by webvictim — APPROVED** [review]
I'm not 100% on board with updating our public-facing Google-indexed docs to recommend that people use an alpha version of Teleport, but I guess we do need to get feedback. As it's in the preview section it should be OK.

**Comment by r0mant** [comment]
@webvictim That's a good point, I've added explicit warnings about these being pre-releases in a couple of places in this document where we say download an alpha.

**Review by benarent — APPROVED** [review]
👍 ~ I've tired this guide and was able to get the alpha up and running.
